base/abstractarray.jl

```diff
@@ -246,8 +246,8 @@ end
 ## Traits for array types ##
 
 abstract LinearIndexing
-immutable LinearFast <: LinearIndexing end
-immutable LinearSlow <: LinearIndexing end
+struct LinearFast <: LinearIndexing end
+struct LinearSlow <: LinearIndexing end
```

Comment:
@StefanKarpinski said:

> So its defining property is that it is made up of fields, not its mutability or lack thereof.

If one agrees with this perspective (I think it makes sense), then structs with no fields feels a little odd.

(I actually thought these lines seemed odd when I ran across them, and post-justified it with Stefan's comment because it fits.)